### PR TITLE
Add function pg_index_column_has_property()

### LINKED ISSUE #2276: Document: feat(expr): add function `pg_index_column_has_property`
This issue tracks the documentation update needed for the merged PR #17275.

Source PR URL: https://github.com/risingwavelabs/risingwave/pull/17275
Source PR Merged At: 2024-06-17T09:09:19Z

### PR DESCRIPTION
**Pull request by ShanlanLi**
## Info

- **Description**

  - [ What's changed? Which parts of the docs are affected? ]

- **Notes**

  - [ Include any supplementary context or references here. ]

- **Related code PR**

  - https://github.com/risingwavelabs/risingwave/pull/17275

- **Related doc issue**
  
  Resolves https://github.com/risingwavelabs/risingwave-docs/issues/2276

## For reviewers

- **Preview**

  - [ Paste the preview link to the updated page(s) here. Edit this item after the preview site is ready. To find the updated pages, scroll down to locate and open the Amplify preview link and select the **dev** version of the documentation. ]

- **Key points**

  - [ Parts that may need revision or extra consideration. ]

## Before merging

- [ ] I have checked the doc site preview, and the updated parts look good.

- [ ] I have acquired the approval from the owner (and optionally the reviewers) of the code PR and at least one tech writer (`emile-00`, `hengm3467`, & `WanYixian`).

### CODE REVIEW

**Comment by ShanlanLi** [comment]
@wangrunji0408 Apart from adding this function, should I add something to remind users we have added new properties to the index catalog? Where may be a good place to do so if it's necessary?

image

**Review by wangrunji0408 — APPROVED** [review]
LGTM!

**Comment by wangrunji0408** [comment]
> @wangrunji0408 Apart from adding this function, should I add something to remind users we have added new properties to the index catalog? Where may be a good place to do so if it's necessary?
> 
> image

No, this is an implementation detail that doesn't need to be exposed to users.

File: docs/sql/functions-operators/sql-function-sys-info.md [hunk 1]
```diff
@@ -126,6 +126,43 @@ SELECT pg_get_viewdef('materialized_view1'::regclass);
 (1 row)
 ```
 
+## `pg_index_column_has_property()`
+
+Checks if an index column has a specific property. The `index` parameter represents the OID of the index, while the `column` parameter represents the column number (starting from 1) within the index. If the property name is not recognized or doesn't apply to the object, or if the OID or column number is invalid, NULL will be returned.
```
Comment (WanYixian) [inline docs/sql/functions-operators/sql-function-sys-info.md]:
```suggestion
Checks if an index column has a specific property. The `index` parameter represents the OID of the index, while the `column` parameter represents the column number (starting from 1) within the index. If the property name is not recognized or doesn't apply to the object, or if the OID or column number is invalid, this function will return NULL.
```

**Review by WanYixian — APPROVED** [review]
others LGTM

**Pull request merged**